platform/linux-generic/pktio/dpdk.c

```diff
@@ -98,19 +98,36 @@ static int lookup_opt(const char *path, const char *drv_name, int *val)
 {
 	const char *base = "pktio_dpdk";
 	char opt_path[256];
-	int ret = 0;
+	int drv_ret = 0;
+	int def_ret = 0;
+	int drv_val = 0;
+	int def_val = 0;
+
+	/* Driver specific option */
+	snprintf(opt_path, sizeof(opt_path), "%s.%s.%s", base, drv_name, path);
+	drv_ret = _odp_libconfig_lookup_int(opt_path, &drv_val);
+	if (drv_ret == LIBCONFIG_OPT_RUNTIME) {
+		*val = drv_val;
```

Comment:
why you modify val is error is returned?

Reply:
same comment for other changes in this file. If val is not needed to modify then def_val and def_ret might be simple ret and rval.

Reply:
This is not error handling. 1 is returned if a value was found.